File: Python/Learn_python/truyvantong.py
def truyvantong(x,y,A):
    sum = 0 
    for i in range(x,y):
        sum += A[i-1]
        
    return sum

n = int(input())

A = []

for i in range(n):
    A.append(int(input()))

x = int(input())

y = int(input())

print(truyvantong(x,y,A))

File: CPP/baitap/truyvantong.cpp
#include<iostream>
#include<vector>
using namespace std;
int truyvantong(int a, int b, vector <int> A){
    int sum = 0;
    for (int i = a; i<=b;i++){
        sum+=A[i-1];
    }
    return sum;
}
int main(){
    int n,q;
    cin>>n>>q;
    vector <int> A;
    for (int i=0;i<n;i++){
        int temp;
        cin>>temp;
        A.push_back(temp);
    }
    int a,b;
    vector <int> B;
    for (int i=0;i<q;i++){
        cin>>a>>b;
        B.push_back(truyvantong(a,b,A));
    }
    for (int i = 0; i<q;i++)
        cout<<B[i]<<endl;
}